core/src/test/scala/org/apache/spark/sql/store/CreateIndexTest.scala

```diff
@@ -0,0 +1,772 @@
+    snContext.sql("drop index test1")
+
+
+    snContext.createIndex("test1", tableName, Map(("col1" -> None)), Map("index_type" -> "unique"))
+    snContext.dropIndex("test1", false)
+    snContext.createIndex("test1", tableName,
+      Map(("col1" -> None)), Map("index_type" -> "global hash"))
+    snContext.dropIndex("test1", false)
+    snContext.createIndex("test1", tableName, Map(("col1" -> None)),
+      Map.empty[String, String])
+    snContext.dropIndex("test1", false)
+
+    snContext.createIndex("test1", tableName,
+      Map(("col1" -> Some(Descending))), Map("index_type" -> "unique"))
+    snContext.dropIndex("test1", false)
+
+    // drop non-existent indexes with if exist clause
+    snContext.dropIndex("test1", true)
+    snContext.sql("drop index if exists test1")
+
+
+    doPrint("Create Index - Start")
+    snContext.sql("create index test1 on " + tableName + " (COL1)")
+    doPrint("Create Index - Done")
+
+    // TODO fails if column name not in caps
+    val result = snContext.sql("select COL1 from " +
+        tableName +
+        " where COL2 like '%a%'")
+    result.explain(true)
+    doPrint("")
+    doPrint("=============== RESULTS START ===============")
+    result.collect.foreach(doPrint)
+    result.collect.foreach(verifyRows)
+    doPrint("=============== RESULTS END ===============")
+
+    doPrint("Drop Index - Start")
+    snContext.sql("drop index test1")
+    doPrint("Drop Index - Done")
+  }
+
+  def verifyRows(r: Row): Unit = {
+    doPrint(r.toString())
+    assert(r.toString() == "[111]", "got=" + r.toString() + " but expected 111")
+  }
+
+  def doPrint(s: Any): Unit = {
+    // println(s)
+  }
+
+  private case class QueryExecutor(snContext: SnappyContext,
+      showResults: Boolean = false,
+      withExplain: Boolean = false) {
+
+    private[store] def apply(sqlText: String, explainQ: Boolean = false)
+        (implicit validate: DataFrame => Unit = _ => ()) = {
+      val msg = s"Executing $sqlText"
+      logInfo(msg) // log it
+      info(msg) // inform it
+
+      val selectRes = snContext.sql(sqlText)
+
+      if (withExplain || explainQ) {
+        selectRes.explain(true)
+      }
+
+      validate(selectRes)
+
+      if (showResults) {
+        selectRes.show
+      } else {
+        // scalastyle:off println
+        selectRes.collect().take(10).foreach(println)
+        // scalastyle:on println
+      }
+
+      selectRes
+    }
+
+  }
+
+  private def validateIndex(index: Seq[String], tables: String*)(df: DataFrame): Unit = {
+    val (indexesMatched, indexesUnMatched) = df.queryExecution.optimizedPlan.collect {
+      case l@LogicalRelation(idx: IndexColumnFormatRelation, _, _) => idx
+    }.partition(rel => index.exists(i => rel.table.indexOf(i.toUpperCase) > 0))
+
+    if (indexesMatched.size != index.size) {
+      df.explain(true)
+      fail(s"Expected index selection ${index.mkString(",")}, but found" +
+          s" ${indexesUnMatched.mkString(",")}")
+    }
```

Comment:
@ymahajan it is doing strict validation based on incoming indexes. here its printing actual names of the indexes matched.
```java
val (indexesMatched, indexesUnMatched) = df.queryExecution.optimizedPlan.collect {
+      case l@LogicalRelation(idx: IndexColumnFormatRelation, _, _) => idx
+    }.partition(rel => index.exists(i => rel.table.indexOf(i.toUpperCase) > 0))
+
```